lib/block-supports/block-style-variations.php

```diff
@@ -79,6 +115,10 @@ function gutenberg_render_block_style_variation_support_styles( $parsed_block )
 		return $parsed_block;
 	}
 
+	// Recursively resolve any ref values with the appropriate value within the
+	// theme_json data.
+	gutenberg_resolve_block_style_variation_ref_values( $variation_data, $theme_json );
```

Comment:
Just checking so I understand the approach: the refs in `$theme_json` have already been resolved via [get_property_value](https://github.com/WordPress/gutenberg/blob/ba1c881a9615a9e6fdfad6b52c761c59a0a77733/lib/class-wp-theme-json-gutenberg.php#L2439-L2439), and you're using the "ref" values in the variation to fetch the resolved values from `$theme_json`?

Reply:
> the refs in $theme_json have already been resolved via [get_property_value](https://github.com/WordPress/gutenberg/blob/ba1c881a9615a9e6fdfad6b52c761c59a0a77733/lib/class-wp-theme-json-gutenberg.php#L2439-L2439)

This works a little differently as `get_property_value` is called when computing theme.json styles. When processing a block style variation, it's a partial set of the overall theme.json data that doesn't always contain the values that might have been referenced. 

So the solution here resolves those values before generating the variation's styles.

Reply:
Thanks, it's been a while since I touched refs. All good 🚀